tests/utils.py

```diff
@@ -0,0 +1,133 @@
+import itertools
+import random
+import string
+
+import avro.schema
+
+
+def valueFromSetGenerator(possibilities=[]):
+    while True:
+        yield random.choice(possibilities)
+
+
+def noneGenerator():
+    while True:
+        yield None
+
+
+def booleanGenerator():
+    gen = valueFromSetGenerator([True, False])
+    while True:
+        yield gen.next()
+
+
+def integerGenerator(min=-2 ** 31, max=2 ** 31):
+    while True:
+        yield random.randint(min, max)
+
+
+def longGenerator(min=-2 ** 63, max=2 ** 63):
+    while True:
+        yield random.randint(min, max)
+
+
+def floatGenerator(min=-2 ** 31, max=2 ** 31):
+    while True:
+        yield random.uniform(min, max)
+
+
+def doubleGenerator(min=-2 ** 63, max=2 ** 63):
+    while True:
+        yield random.uniform(min, max)
+
+
+def stringGenerator(length=10, characters=string.printable):
+    gen = valueFromSetGenerator(characters)
+    while True:
+        yield "".join(itertools.islice(gen, length))
+
+
+def unionGenerator(generators):
+    while True:
+        yield random.choice(generators).next()
+
+
+def arrayGenerator(generator, length=10):
+    while True:
+        yield list(itertools.islice(generator, length))
+
+
+def mapGenerator(valueGen, size=10, keyGen=stringGenerator()):
+    while True:
+        keys = list(itertools.islice(keyGen, size))
+        values = list(itertools.islice(valueGen, size))
+        yield dict(zip(keys, values))
+
+
+def recordGenerator(recordSchema):
+    while True:
+        record = {}
+        for key, gen in recordSchema.iteritems():
+            record[key] = gen.next()
+        yield record
+
+
+class InstanceGenerator:
+    typeToGenerator = {
+        "null": noneGenerator(),
+        "boolean": booleanGenerator(),
+        "int": integerGenerator(),
+        "long": longGenerator(),
+        "string": stringGenerator(),
+        "float": floatGenerator(),
+        "double": doubleGenerator()
+    }
+
+    def __init__(self):
+        self.typeToGenerator = dict(InstanceGenerator.typeToGenerator)
```

Comment:
Why are we allocating a per-instance version of `typeToGenerator`? Will a class variable not work?